pytorch_lightning/trainer/properties.py

```diff
@@ -412,6 +421,76 @@ def distributed_sampler_kwargs(self) -> Optional[dict]:
         if isinstance(self.training_type_plugin, ParallelPlugin):
             return self.training_type_plugin.distributed_sampler_kwargs
 
+    @property
+    def training(self) -> bool:
+        return self._running_stage == RunningStage.TRAINING
+
+    @training.setter
+    def training(self, val: bool) -> None:
+        if val:
+            self._running_stage = RunningStage.TRAINING
+        elif self.training:
+            self._running_stage = None
+
+    @property
+    def testing(self) -> bool:
+        return self._running_stage == RunningStage.TESTING
+
+    @testing.setter
+    def testing(self, val: bool) -> None:
+        if val:
+            self._running_stage = RunningStage.TESTING
+        elif self.testing:
+            self._running_stage = None
+
+    @property
+    def predicting(self) -> bool:
+        return self._running_stage == RunningStage.PREDICTING
+
+    @predicting.setter
+    def predicting(self, val: bool) -> None:
+        if val:
+            self._running_stage = RunningStage.PREDICTING
+        elif self.predicting:
+            self._running_stage = None
+
+    @property
+    def tuning(self) -> bool:
+        return self._running_stage == RunningStage.TUNING
+
+    @tuning.setter
+    def tuning(self, val: bool) -> None:
+        if val:
+            self._running_stage = RunningStage.TUNING
+        elif self.tuning:
+            self._running_stage = None
+
+    @property
+    def validating(self) -> bool:
+        return self._running_stage == RunningStage.VALIDATING
+
+    @validating.setter
+    def validating(self, val: bool) -> None:
+        if val:
+            self._running_stage = RunningStage.VALIDATING
+        elif self.validating:
+            self._running_stage = None
+
+    @property
+    def evaluating(self) -> bool:
+        return self._running_stage and self._running_stage.evaluating
+
+    @property
+    def sanity_checking(self) -> bool:
+        return self._running_stage == RunningStage.SANITY_CHECKING
+
+    @sanity_checking.setter
+    def sanity_checking(self, val: bool) -> None:
+        if val:
+            self._running_stage = RunningStage.SANITY_CHECKING
```

Comment:
Can you rename `SANITY_CHECKING` to `SANITY_VALIDATION_CHECKING ` ? We plan to add  `SANITY_PREDICT_CHECKING `

Reply:
Actually, this refactor enables that seamlessly as `SANITY_CHECKING` is just a RunningStage, so it can be used for both fitting and predicting without needing a different attribute